### PR TITLE
Updating spm syntax to new swift-tools

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
Just updating dependencies to Rx 4.5.0 :))

### CODE REVIEW

**Comment by bobgodwinx** [comment]
@LucianoPAlmeida 
1. If we change the pod spec from `"RxSwift", "~> 4.0` to `"RxSwift", "~> 4.5.0"` it means that you need to have `"RxSwift", "~> 4.5.0"` and above to use the library. Which is not fair because there are people using the previous versions like `"RxSwift", "~> 4.3.0"` etc.. So I think we should leave it the way it was before. 

2. I fail to see the reason why we should not expose the action properties? By moving them from `Public` they become hidden and inaccessible.

**Comment by LucianoPAlmeida** [comment]
@bobgodwinx 
Answering:
1. Didn't thought about that, reverting :)
2. If the extension the property or method is declared is `public extension` the modifier in the function is redundant and the compiler give us warnings :))

**Comment by bobgodwinx** [comment]
@LucianoPAlmeida  sorry for the late reply. Yes the `Public extension` is fine I tested that and it works. However the pod has to actually look like this for it to work properly `"RxSwift", "~> 4.0` `"RxCocoa", "~> 4.0`  so just remove the last zero and we're good to go thanks

File: Action.podspec [hunk 1]
```diff
@@ -20,8 +20,8 @@ Pod::Spec.new do |s|
   s.source_files  = "Sources/**/*.{swift}"
 
   s.frameworks  = "Foundation"
-  s.dependency "RxSwift", "~> 4.0"
-  s.dependency "RxCocoa", "~> 4.0"
+  s.dependency "RxSwift", "~> 4.0.0"
```
Comment (LucianoPAlmeida) [inline Action.podspec]:
```suggestion
  s.dependency "RxSwift", "~> 4.0"
```

File: Action.podspec [hunk 2]
```diff
@@ -20,8 +20,8 @@ Pod::Spec.new do |s|
...
+  s.dependency "RxCocoa", "~> 4.0.0"
```
Comment (LucianoPAlmeida) [inline Action.podspec]:
```suggestion
  s.dependency "RxCocoa", "~> 4.0"
```

**Pull request merged**